### PR TITLE
Merging libcxxthrd to libcxx

### PR DESCRIPTION
**Pull request by anitagov**
Merging libcxxthrd to libcxx (Issue #955). The major changes are as follows:
1) Increased TCS count of libcxx from 2 to 8.
2) Increased heap size of libcxxthrd from 8192 to 12288.
3) All the libcxxthrd changes to enc.cpp, host.cpp and threadArgs.h (new file) are now in the libcxx directory - notably the hooks for pthread_create, pthread_join and pthread_detach.
4) The tests lists were merged - 51 threading tests from libcxx that were unsupported are now supported.
5) The directory libcxxthrd no longer exists (this is the obvious change)

### CODE REVIEW

**Comment by anitagov** [comment]
bors try

**Comment by anitagov** [comment]
bors try

**Comment by anitagov** [comment]
bors try

**Comment by anitagov** [comment]
bors try

File: tests/libcxx/enc/main.cpp [hunk 1]
```diff
@@ -1,5 +1,8 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+// This file exists so that each test has its own copy of
```
Comment (andschwa) [inline tests/libcxx/enc/main.cpp]: Thanks 😄

**Pull request merged**